### PR TITLE
merging sever.js changes

### PR DESCRIPTION
**Pull request by karadouville**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -80,7 +86,48 @@ app.post('/game', validate_create_game, function(req, res) {
     console_log("game {0} created by {1}".format(game.id, game.player1.id));
 });
 
+
 // GET game
+app.get('/setup', function(req, res, next) {
```
Comment (Austinpayne) [inline server.js:91]: @karadouville What is this endpoint used for? Can you just use `post /game` and `post /game/:id/join`? Seems strange to use `get` for creating a new game.

File: server.js [hunk 2]
```diff
@@ -190,15 +237,11 @@ app.post('/game/:id/player/:pid/move', validate_gid, validate_pid,
...
-            res.status(200).json(move); // send back move
-            games[game_id].last_move = Date.now();
-            console.log(chess.ascii());
-            return;
-        }
-
-        res.status(200).json({error: "invalid move"});
+        // we won't check to see if the move was correct or not. 
+        // Just send back the game state after the move was made.
+        res.status(200).json(games[game_id]); // send back move
```
Comment (Austinpayne) [inline server.js:242]: @karadouville Why did you remove the move validation here? Some clients cannot validate moves so we need to have the server validate.

**Comment by Austinpayne** [comment]
@karadouville Looks good to me, but can you rebase onto the current master and resolve my other comments? I think we should either get rid of the `/setup` endpoint or break out the functionality in `/game` and `/game/:id/join` into separate functions that can be called so we can reduce duplication in `/setup`

File: server.js (hunk 1 above)
Comment (karadouville) [reply server.js:91]: @Austinpayne  this endpoint is hit by the login page, I guess it's part of the UI api, rather than the REST api. You're totally right, It is a duplication of /game and /game:id/join, something that I was planning on fixing down the line. I was just trying to crank something out and I needed a different response.  I'm working on extracting the logic from creating a game and joining a game and getting rid of the get setup.

File: server.js (hunk 2 above)
Comment (karadouville) [reply server.js:242]: @Austinpayne sorry, I was going to talk with you about this and forgot. I didn't realize that some clients would be able to validate the moves and that was necessary for the server. For my purpose, it made sense to have the resulting game state returned. We'll chat more about this tomorrow.

Comment (Austinpayne) [reply server.js:242]: I think in this case where you want the game back I would just return it as well with the move object. The nice thing about JSON is you can just add to as you need but replacing here is problematic in this case. Also, since your client already does pre-validation before even sending a move to the server I think we can still leave the "error" case in. Let me know what you think.

File: server.js (hunk 1 above)
Comment (Austinpayne) [reply server.js:91]: Yeah, just seems to be blurring the lines a lot. I'm fine with it, just wasn't sure what it is for. Where we are mixing the backend with frontend we should at least put it in a different spot and have some comment header marking it as the "UI" section.

**Comment by Austinpayne** [comment]
moving to new PR
